toml/src/test/kotlin/org/rust/toml/completion/LocalCargoTomlDependencyCompletionProviderTest.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml.completion
+
+class LocalCargoTomlDependencyCompletionProviderTest : LocalCargoTomlCompletionTestBase() {
+    fun `test basic completion`() = doSingleCompletion("""
+        [dependencies]
+        fo<caret>
+    """, """
+        [dependencies]
+        foo = "1.0.0"<caret>
+    """, crate("foo", "1.0.0"), crate("bar", "1.0.0"))
+
+    fun `test no completion`() = checkNoCompletion("""
+        [dependencies]
+        fo<caret>
+    """, crate("bar", "1.0.0"))
+
+    fun `test hyphen-underscore normalization completion`() = doSingleCompletion("""
+        [dependencies]
+        foo-<caret>
+    """, """
+        [dependencies]
+        foo_bar = "1.0.0"<caret>
+    """, crate("foo_bar", "1.0.0"))
+
+    fun `test subwords completion`() = doSingleCompletion("""
+        [dependencies]
+        f-ba<caret>
+    """, """
+        [dependencies]
+        foo_bar = "1.0.0"<caret>
+    """, crate("foo_bar", "1.0.0"))
+}
```

Comment:
Maybe provide some tests with multiple versions of the same crate to check that the latest version is inserted?

Reply:
Right now it wouldn't be that useful, as without semantic versioning sorting it just takes the last one, but I did add one.

Also, I'm still thinking that maybe we should insert the name only in the name completion to decrease the number of situations when completion leads to invalid code. Then, such tests will be performed in version intention/completion tests